client.go

```diff
@@ -53,27 +63,44 @@ type Client struct {
 //
 // If your use case involves multiple long-lived connections, consider using
 // the ClientManager, which manages clients for you.
-func NewClient(certificate tls.Certificate) *Client {
+//
+// Alternatively, you can keep the clients connection healthy by calling
+// EnablePinging, which will send PING frames to APNs servers with the interval
+// specified via PingPongFrequency.
+func NewClient(certificate tls.Certificate) (client *Client) {
 	tlsConfig := &tls.Config{
 		Certificates: []tls.Certificate{certificate},
 	}
 	if len(certificate.Certificate) > 0 {
 		tlsConfig.BuildNameToCertificate()
 	}
+	client = &Client{
+		Certificate: certificate,
+		Host:        DefaultHost,
+		newConnChan: make(chan struct{}),
+		stopChan:    make(chan struct{}),
+		m:           new(sync.Mutex),
+	}
 	transport := &http2.Transport{
 		TLSClientConfig: tlsConfig,
-		DialTLS: func(network, addr string, cfg *tls.Config) (net.Conn, error) {
-			return tls.DialWithDialer(&net.Dialer{Timeout: TLSDialTimeout}, network, addr, cfg)
+		DialTLS: func(network, addr string, cfg *tls.Config) (c net.Conn, e error) {
+			c, e = tls.DialWithDialer(&net.Dialer{Timeout: TLSDialTimeout}, network, addr, cfg)
+			if e == nil {
+				client.m.Lock()
+				defer client.m.Unlock()
+				client.conn = c
+				if client.pinging {
+					client.newConnChan <- struct{}{}
```

Comment:
It shouldn't since when pinging is false, the function won't try to send a signal to newConnChan, and releases the lock when it returns. When pinging is true, there is a goroutine that reads from newConnChan, so writing to it should never block. Am I missing something?